qt/main.cpp

```diff
@@ -213,7 +203,32 @@ int main(int argc, char * argv[])
         screenshotParams->m_dpiScale = FLAGS_dpi_scale;
     }
 
-    qt::MainWindow::SetDefaultSurfaceFormat(apiOpenGLES3);
+
+    QSurfaceFormat fmt;
+    fmt.setAlphaBufferSize(8);
+    fmt.setBlueBufferSize(8);
+    fmt.setGreenBufferSize(8);
+    fmt.setRedBufferSize(8);
+    fmt.setStencilBufferSize(0);
+    fmt.setSamples(0);
+    fmt.setSwapBehavior(QSurfaceFormat::DoubleBuffer);
+    fmt.setSwapInterval(1);
+    fmt.setDepthBufferSize(16);
+#if defined(OMIM_OS_LINUX)
+    // This is a workaround for older distros, which rely on X.org and Mesa,
+    // where somehow the Mesa driver itself doesn't
+    // make all the otherwise supported GLSL versions available by default
+    // and such requests are somehow disregarded at later stages of execution.
+    // This setting here will be potentially overwritten and overruled anyway,
+    // and only needed to ensure that we have the needed GLSL compiler available
+    // later when we need it.
+    if (app.platformName() == QString("xcb"))
+      fmt.setVersion(3, 2);
+#endif
```

Comment:
@Ferenc- sorry if it was discussed earlier, could you expand the problem to me?
What versions get applied by default without this patch, for XCB and Cocoa?